### PR TITLE
Fix remove unexpected whitespaces introduced by formatting

### LINKED ISSUE #2050: Support block judoc comments
As we have `--` for single line comments and `{-` `-}` for multiline block comments, it would be convenient to support `{--` `--}` block Judoc comment with the existing single line `---`.

### LINKED ISSUE #2059: Unexpected whitespace introduced by the formatter
I'm getting extra spaces when running the formatter, for example with the
Bank example. Run the following to reproduce the error:

```bash
juvix format --in-place examples/milestone/Bank/Bank.juvix
```

https://github.com/anoma/juvix/blob/8f850c2e5b56182f4ce375245298ea5d77360bd8/examples/milestone/Bank/Bank.juvix#L69

```diff
  decrement _ _ nil := nil;
  decrement f n ((b, bn) :: bs) :=
    if
      (eqField f b)
      ((b, sub bn n) :: bs)
      ((b, bn) :: decrement f n bs);
-
+  
  emtpyBalances : Balances;
  emtpyBalances := nil;
```

### PR DESCRIPTION
**Pull request by jonaprieto**
- Closes #2059

### CODE REVIEW

File: tests/positive/Format.juvix [hunk 1]
```diff
@@ -188,19 +188,22 @@ f : Nat -> Nat :=
   };
 
 module Patterns;
+
```
Comment (jonaprieto) [inline tests/positive/Format.juvix:191]: Added spaces on this file on purpose.

File: src/Juvix/Prelude/Pretty.hs [hunk 2]
```diff
@@ -24,15 +24,18 @@ defaultLayoutOptions =
...
   toAnsiStream :: a -> SimpleDocStream Ansi.AnsiStyle
-  toAnsiStream = layoutPretty defaultLayoutOptions . toAnsiDoc
+  toAnsiStream = rTW . toAnsiDoc
 
   toAnsiDoc :: a -> Doc Ansi.AnsiStyle
 
 class HasTextBackend a where
   toTextStream :: a -> SimpleDocStream b
-  toTextStream = layoutPretty defaultLayoutOptions . toTextDoc
+  toTextStream = rTW . toTextDoc
```
Comment (paulcadman) [inline src/Juvix/Prelude/Pretty.hs:38]: This change applies to all text output, and so bypasses the `format: false` pragma (i.e modules that have `format: false` set will still have whitespace trimming).

File: src/Juvix/Prelude/Pretty.hs [hunk 3]
```diff
@@ -157,31 +163,31 @@ hsepMaybe l
   | null l = Nothing
   | otherwise = Just (hsep l)
 
-nest' :: Doc ann -> Doc ann
+nest' :: Doc a -> Doc a
```
Comment (janmasrovira) [inline src/Juvix/Prelude/Pretty.hs:166]: do you want to keep this change?

File: src/Juvix/Prelude/Pretty.hs [hunk 4]
```diff
@@ -123,6 +126,9 @@ toAnsiText useColors
 toPlainText :: (HasTextBackend a) => a -> Text
 toPlainText = Text.renderStrict . toTextStream
 
+toPlainTextTrim :: (HasTextBackend a) => a -> Text
+toPlainTextTrim = Text.renderStrict . rTW . toTextDoc
```
Comment (janmasrovira) [inline src/Juvix/Prelude/Pretty.hs:130]: why not rendering to text and then removing the trailing whitespace as we discussed?

**Review by janmasrovira — APPROVED** [review]

**Pull request merged**